dask/utils.py

```diff
@@ -1229,6 +1229,18 @@ def get_scheduler_lock(collection=None, scheduler=None):
 
     if actual_get == multiprocessing.get:
         return multiprocessing.get_context().Manager().Lock()
+    else:
+        # if this is a distributed client with processes, we also need to lock
+        try:
+            from distributed.utils import get_mp_context
+            from distributed.worker import get_client
+
+            client = get_client()
+        except (ImportError, ValueError):
+            pass
+        else:
+            if actual_get == client.get and getattr(client.cluster, "processes", None):
+                return get_mp_context().Manager().Lock()
```

Comment:
I think this should use a `distributed.Lock` so it works on multi-machine clusters, although I'm not sure if it's going to be reliable to run to_hdf on one file with a multi machine cluster on a network filesystem

Reply:
I didn't dive deep into the code, but I agree with @graingert that you probably want to use a `distributed.Lock` when dealing with locking on a `distributed` cluster. Let me know if I can help you with this.

Reply:
@hendrikmakait If you are able to help, it would be great. I don't know how to correctly check if the cluster uses processes ([this comment](https://github.com/dask/dask/pull/10309/files#r1205346394)). Can you advise?